src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -431,5 +438,52 @@ public async Task<string> GetRepositoryPolicies(string repoUri, string branchNam
         {
             return await RunDarcAsync("get-repository-policies", "--all", "--repo", repoUri, "--branch", branchName);
         }
+        public async Task<bool> WaitForMergedPullRequestAsync(string targetRepo, string targetBranch, PullRequest pr, Repository repo, int attempts = 7)
+        {
+            while (attempts-- > 0)
+            {
+                TestContext.WriteLine($"Starting check for merge, attempts remaining {attempts}");
+                pr = await GitHubApi.PullRequest.Get(repo.Id, pr.Number);
+
+                if (pr.State == ItemState.Closed)
+                {
+                    return true;
+                }
+
+                await Task.Delay(60 * 1000).ConfigureAwait(false);
+            }
+
+            throw new MaestroTestException($"The created pull request for {targetRepo} targeting {targetBranch} was not merged within {attempts} minutes");
+        }
+
+        public async Task<bool> CheckGithubPullRequestChecks(string targetRepoName, string targetBranch)
+        {
+            TestContext.WriteLine($"Checking opened PR in {targetBranch} {targetRepoName}");
+            PullRequest pullRequest = await WaitForPullRequestAsync(targetRepoName, targetBranch);
+            Repository repo = await GitHubApi.Repository.Get(_parameters.GitHubTestOrg, targetRepoName).ConfigureAwait(false);
+
+            return await ValidateGithubMaestroCheckRunsSuccessful(targetRepoName, targetBranch, pullRequest, repo);
+        }
+
+        public async Task<bool> ValidateGithubMaestroCheckRunsSuccessful(string targetRepoName, string targetBranch, PullRequest pullRequest, Repository repo)
+        {
+            // Waiting 5 minutes for maestro to add the checks to the PR
+            await Task.Delay(300000);
```

Comment:
await Task.Delay(TimeSpan.FromMinutes(5)) is a lot easier to not typo (I know I'm the worst for having one too many or two few zeros), and to code review.